rgf/test/test.py

```diff
@@ -72,21 +77,65 @@ def test_regressor(self):
         assert_less(mse, 6.0)
 
     def test_sample_weight(self):
-        X_train, X_test, y_train, y_test = train_test_split(self.iris.data,
-                                                            self.iris.target,
-                                                            test_size=0.2,
-                                                            random_state=42)
-
         clf = RGFClassifier()
         
-        y_pred = clf.fit(X_train, y_train).predict_proba(X_test)
-        y_pred_weighted = clf.fit(X_train, y_train, np.ones(y_train.shape[0])).predict_proba(X_test)
+        y_pred = clf.fit(self.X_train, self.y_train).predict_proba(self.X_test)
+        y_pred_weighted = clf.fit(self.X_train,
+                                  self.y_train,
+                                  np.ones(self.y_train.shape[0])).predict_proba(self.X_test)
         np.testing.assert_allclose(y_pred, y_pred_weighted)
 
-        weights = np.ones(y_train.shape[0]) * np.nextafter(np.float32(0), np.float32(1))
+        weights = np.ones(self.y_train.shape[0]) * np.nextafter(np.float32(0), np.float32(1))
         weights[0] = 1
-        y_pred_weighted = clf.fit(X_train, y_train, weights).predict(X_test)
-        np.testing.assert_equal(y_pred_weighted, np.full(y_test.shape[0], y_test[0]))
+        y_pred_weighted = clf.fit(self.X_train, self.y_train, weights).predict(self.X_test)
+        np.testing.assert_equal(y_pred_weighted, np.full(self.y_test.shape[0], self.y_test[0]))
+
+    def test_params(self):
+        clf = RGFClassifier()
+        
+        valid_params = dict(max_leaf=1100,
+                            test_interval=100,
+                            algorithm='RGF_Sib',
+                            loss='Log',
+                            reg_depth=1.1,
+                            l2=0.1,
+                            sl2=None,
+                            normalize=False,
+                            min_samples_leaf=9,
+                            n_iter=None,
+                            n_tree_search=2,
+                            opt_interval=100,
+                            learning_rate=0.4,
+                            verbose=True,
+                            prefix='rgf_classifier',
+                            inc_prefix=True,
+                            calc_prob='Sigmoid',
+                            clean=True)
+        clf.set_params(**valid_params)
+        y_pred = clf.fit(self.X_train, self.y_train).predict(self.X_test)
```

Comment:
We don't need y_pred.
`clf.fit(self.X_train, self.y_train).predict(self.X_test)` only is OK.